src/test/kotlin/org/rust/ide/actions/mover/RsStatementUpDownMoverTest.kt

```diff
@@ -0,0 +1,403 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.actions.mover
+
+class RsStatementUpDownMoverTest : RsStatementUpDownMoverTestBase() {
+    fun `test straightline down`() = moveDown("""
+        fn main() {
+            1;
+            2/*caret*/;
+            3;
+        }
+    """, """
+        fn main() {
+            1;
+            3;
+            2/*caret*/;
+        }
+    """)
+
+    fun `test straightline up`() = moveUp("""
+        fn main() {
+            1;
+            2;
+            3/*caret*/;
+        }
+    """, """
+        fn main() {
+            1;
+            3/*caret*/;
+            2;
+        }
+    """)
+
+    fun `test straightline down out of body`() = moveDown("""
+        fn main() {
+            1;
+            2;
+            3/*caret*/;
+        }
+    """, """
+        fn main() {
+            1;
+            2;
+            3/*caret*/;
+        }
+    """)
```

Comment:
This repetition of test data seems unfortunate. Let's add a helper `moveDownForbidden` or something, which will accept only single text?

Reply:
Done